ergo-chain-sync/src/client/node.rs

```diff
@@ -49,4 +51,23 @@ impl ErgoNetwork for ErgoNodeHttpClient {
         }
         None
     }
+
+    async fn get_ergo_state_context(&self) -> Option<ErgoStateContext> {
```

Comment:
Need `ErgoStateContext` for transaction signing. I'm not sure how you want to implement periodic refresh of this data. Just let me know.

Reply:
We could make `SigmaProver.sign()` async and keep `cached_ctx: Cell<(ErgoStateContext, CtxCachedAt)>` in Wallet. Then each time sign is called we could read from the cell and if `timestamp_now - ctx_cached_at > ctx_cache_ttl` then pull new context from the network and rewrite the cell, otherwise just use the cached value.
But no need for that in our case.

Reply:
I like this approach. Write it once and you won't have to worry about stale headers anymore. Shall I implement this?

Reply:
In our case all we get from this will be redundant overhead, bc we only sign p2pk inputs. We can literarily put anything in that context.

Reply:
What are some examples where the context would matter?

Since it's not needed, will also remove `ErgoNetwork::get_ergo_state_context`

Reply:
Haven't seen such cases in practice, but I imagine it is needed when proving some complex sigma propositions that not only involve ProveDLog but also computations that involve blockchain context.

Reply:
I see, thanks.